awx/main/tests/functional/test_inventory_source_injectors.py

```diff
@@ -193,7 +193,7 @@ def create_reference_data(source_dir, env, content):
 
 
 @pytest.mark.django_db
-@pytest.mark.parametrize('this_kind', CLOUD_PROVIDERS)
+@pytest.mark.parametrize('this_kind', discover_available_cloud_provider_plugin_names())
```

Comment:
Let's try this instead so that the unnecessary test isn't produced in the first place:
```suggestion
@pytest.mark.parametrize('this_kind', [plugin_name for plugin_name in discover_available_cloud_provider_plugin_names() if plugin_name != 'constructed'])
```